# Awesome phages

A curated list of awesome phage related software and other resources to facilitate the life of phage scientists,
bioinformaticians and enthusiasts.


## How to contribute

If you want to add a link, please try to follow these soft rules:

- Position your resource in alphabetical order.
- Add section if needed.
- Add section title to Table of contents.
- Search previous suggestions before making a new one, as yours may be a duplicate.
- Add your links using another entry as inspiration
- Check your spelling and grammar.
- Make sure your text editor is set to remove trailing whitespace.
- Send a Pull Request.
- Add short description in your Pull Request of why the project/resource/tool is awesome.

## Table of Contents

- [Annotation](#annotation)
  - [Genome prediction](#genome-prediction) 
  - [Host prediction](#host-prediction)
  - [Prophage prediction](#prophage-prediction)
  - [Gene prediction](#gene-prediction)
- [Web tools](#web-tools)
- [Miscellaneous](#miscellaneous)

## Annotation

### Genome prediction

- **[Seeker](https://github.com/gussow/seeker)** - Seeker is a python library for discriminating between bacterial and
phage genomes. Seeker is based on an LSTM deep-learning models and does not rely on a reference genome, genomic
alignment or any direct genome comparison.
[ [web](http://seeker.pythonanywhere.com/predict/) ]
- **[VIRify](https://github.com/EBI-Metagenomics/emg-viral-pipeline)** - VIRify: detection of phages and eukaryotic
viruses from metagenomic and metatranscriptomic assemblies.

### Host prediction

- **[phirbo](https://github.com/aziele/phirbo)** - A tool to predict prokaryotic hosts for phage (meta)genomic
sequences. The tool uses information on sequence similarity between phage and bacteria genomes as well as sequence
relatedness among bacteria.

### Prophage prediction

- **[DBSCAN-SWA](https://github.com/HIT-ImmunologyLab/DBSCAN-SWA)** - an integrated tool for rapid prophage detection
and annotation.
- **[hafeZ](https://github.com/Chrisjrt/hafeZ)** - A tool for identifying inducible prophage elements through read
mapping.
- **[Phigaro](https://github.com/bobeobibo/phigaro)** - Phigaro is a standalone command-line application that is able
to detect prophage regions taking raw genome and metagenome assemblies as an input.
- **[PhageBoost](https://github.com/ku-cbd/PhageBoost)** - Rapid discovery of novel prophages using biological feature
engineering and machine learning.
- **[PhiSpy](https://github.com/linsalrob/phispy)** - PhiSpy identifies prophages in Bacterial genomes. Given an
annotated genome it will use several approaches to identify the most likely prophage regions.
- **[Vibrant](https://github.com/AnantharamanLab/VIBRANT)** - VIBRANT is a tool for automated recovery and annotation
of bacterial and archaeal viruses, determination of genome completeness, and characterization of viral community
function from metagenomic assemblies.
- **[VirSorter2](https://github.com/jiarong/VirSorter2)** - VirSorter2 applies a multi-classifier, expert-guided
approach to detect diverse DNA and RNA virus genomes.

### Gene prediction

- **[multiPhATE2](https://github.com/carolzhou/multiPhATE2)** - MultiPhATE is a command-line program that runs gene
finding and the PhATE annotation code over user-specified phage genomes, then performs gene-by-gene comparisons among
the genomes.
- **[Phanns](https://github.com/Adrian-Cantu/PhANNs)** - PhANNs is a tool to classify any phage ORF as one of 10
structural protein class, or as "others". It uses an ensemble of Artificial Neural Networks.
[ [web](https://edwards.sdsu.edu/phanns) ]
- **[Phanotate](https://github.com/deprekate/PHANOTATE)** - PHANOTATE is a tool to annotate phage genomes. It uses the
assumption that non-coding bases in a phage genome is disadvantageous, and then populates a weighted graph to find the
optimal path through the six frames of the DNA where open reading frames are beneficial paths, while gaps and overlaps
are penalized paths.
- **[Prokka](https://github.com/tseemann/prokka)** - Prokka is not phage specific but is often used to annotate phages.

## Web tools

- **[Phaster](https://phaster.ca/)** - Rapid identification and annotation of prophage sequences within bacterial
genomes and plasmids.
- **[Patric](https://www.patricbrc.org/)** PATRIC, the Pathosystems Resource Integration Center, provides integrated
data and analysis tools to support biomedical research on bacterial infectious diseases.
- **[Phamerator](https://phamerator.org/)** - Comparative genomics and genome exploration tool for bacteriophages.
[ [source](https://github.com/scresawn/phamerator) ]
- **[RAST](https://rast.nmpdr.org/)** - RAST (Rapid Annotation using Subsystem Technology) is a fully-automated service
for annotating complete or nearly complete bacterial and archaeal genomes. It provides high quality genome annotations
for these genomes across the whole phylogenetic tree.

##  Miscellaneous

- **[inphared](https://github.com/RyanCook94/inphared)** - Providing up-to-date phage genome databases, metrics and
useful input files for a number of bioinformatic pipelines.
- **[millardlab](http://millardlab.org/)** - Lab from the University of Leicester focused on bacteriophages with plenty
of resources for phage bioinformatics.
- **[phage.directory](https://phage.directory/)** - Phage Directory curates a database of phage labs, phages, and host
strains to advance research and phage therapy.
- **[phagesdb](https://phagesdb.org/)** - The Actinobacteriophage database, more commonly known as PhagesDB, is a
database-backed website that gathers and shares information related to the discovery, characterization and genomics of
viruses that prefer to infect Actinobacterial hosts
- **[PHAGES2050](https://github.com/ptynecki/PHAGES2050)** - a comprehensive AI-based Framework for Phage Research & Therapy
- **[phage.ai](https://phage.ai/)** - PhageAI is an AI-driven software platform using advanced Machine Learning and
Natural Language Processing techniques for deeper understanding of the bacteriophages genomics.
- **[seaphages](https://seaphages.org/)** - SEA-PHAGES is a two-semester, discovery-based undergraduate research course
that begins with simple digging in the soil to find new viruses, but progresses through a variety of microbiology
techniques and eventually to complex genome annotation and bioinformatic analyses.


## Kwnologe base

- **[Viral taxonomy](https://talk.ictvonline.org/ictv-reports/ictv_online_report/)**
